api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -86,20 +86,18 @@ public void updateUserRoles(String ns, String id, Set<WorkspaceUserRole> userRol
     Map<Integer, WorkspaceUserRole> userRoleMap = new HashMap<Integer, WorkspaceUserRole>();
     for (WorkspaceUserRole userRole : userRoleSet) {
       userRole.setWorkspace(dbWorkspace);
-      int key = Objects.hash(userRole.getUser().getUserId(), userRole.getWorkspace().hashCode());
+      int key = Objects.hash(userRole.getUser().getUserId());
```

Comment:
Don't use Objects.hash() here. Objects.hash() can have collisions.

Just use getUserId(), which is already an integer.